### DISCUSSION TITLE
.zip handling

### DISCUSSION THREAD

**Discussion by Terroosh** [BlafKing/sd-civitai-browser-plus#43] [Ideas]
There are some models on civitai that are packaged in a .zip file such as https://civitai.com/models/143670/peach-green-photography

Currently sd-civitai-browser-plus downloades the zip file and leaves that in the lora folder. Could handling be added to automatically extract the .zip file?

**Comment by BlafKing** [comment]
That should definitely be possible!
I'll probably make it so that there will be an extra setting in the CivitAI settings tab where you can toggle automatic unpacking of .zip downloads!

I'll try to implement this soon, shouldn't be too hard to do :)

**Comment by BlafKing** [comment]
Just published an update with a setting toggle to allow automatically unpacking .zip models in v1.12.3!

Took a bit longer than intended because the default `zipfile` module from Python had trouble unpacking files with Chinese characters since they used different encodings, it should now be able to detect the correct encoding and unpack it properly though! :)
